BUILDING.md specifies Qt v6.9.1 for macOS but Qt v6.10.x required

### Bug description

BUILDING.md lists "Qt 6.9.1 'Desktop' (macOS)" as the required version for that OS, but building with Qt 6.9.x fails with a compilation error: "tablesortfilterproxymodel.cpp:123:21: error: no member named 
  'Direction' in 'QSortFilterProxyModel'". This is because QSortFilterProxyModel::Direction was introduced in Qt 6.10 ([QSortFilterProxyModel Class](https://doc.qt.io/qt-6/qsortfilterproxymodel.html#Direction-enum).)





### Steps to reproduce

1. Follow BUILDING.md instructions and install QT 6.9.1 (or 6.9.3)
2. cmake -S . -B build/
3. cmake --build build/ 
4. Build fails: tablesortfilterproxymodel.cpp:123:21: error: no member named 
  'Direction' in 'QSortFilterProxyModel'

### Expected behavior

BUILDING.md needs to list Qt 6.10 as the minimum requirement.

### Actual behavior

BUILDING.md lists: Qt 6.9.1 'Desktop' (macOS).

### Audacity Version

4.0.0 (master, commit 904f6a3)

### Operating system

macOS

### Additional context

".../GitHub/audacity/muse/framework/uicomponents/qml/Muse/UiComponents/tablesortfilterproxymodel.cpp:123:21: error: 
      no member named 'Direction' in 'QSortFilterProxyModel'; did you mean
      'sigaction'?
  123 |     endFilterChange(QSortFilterProxyModel::Direction::Rows);"

<img width="865" height="636" alt="Image" src="https://github.com/user-attachments/assets/c4c6f95c-d5ae-4ef1-8743-399fbb4e0c3b" />

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -9,7 +9,7 @@
 * A package manager (tested: Choco on Windows, Homebrew on macOS)
 * A CMake generator (tested: Ninja)
 * A C++ compiler (tested: MSVC on Windows, g++ on Linux)
-* Qt 6.9.1 'Desktop' (macOS), 'MSVC 2022 64-bit' (Windows) or 'MSVC 2022 ARM64' (Windows on ARM) with 'Additional Libraries':
+* Qt 6.10 'Desktop' (macOS), 'MSVC 2022 64-bit' (Windows) or 'MSVC 2022 ARM64' (Windows on ARM) with 'Additional Libraries':
   * Qt 5 Compatibility Module
   * Qt Network Authorization
   * Qt Shader Tools

